File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int size = nums.length;
        int[] ans = new int[size];
        
        int prefix = 1;
        for (int i = 0; i < size; i++){
            ans[i] = prefix;
            prefix = prefix * nums[i];
        }
        
        int postfix = 1;
        for (int j = size-1; j >= 0; j--){
            ans[j] = ans[j] * postfix;
            postfix = postfix * nums[j];
        }
        
        return ans;
        
        
        
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        res = []
        
        prefix = 1
        for i in range(len(nums)):
            res.append(prefix)
            prefix *= nums[i]
            
        postfix = 1
        for i, elem in reversed(list(enumerate(nums))):
            res[i] = postfix * res[i]
            postfix = postfix * nums[i]
            
        return res
            
            
        